src/code/views/app-view.coffee

```diff
@@ -74,14 +55,30 @@ module.exports = React.createClass
 
   onLinkChanged: (link, title, color, deleted) ->
     @props.linkManager.changeLink title,color, deleted
-  
+
+  onAddRemoteImage: (image) ->
+    # make sure this is a new image
+    if not _.find @state.protoNodes, {image: image}
```

Comment:
Do you think it might time to refactor?  I suspect that this logic could be more lucidly expressed in a thicker proto-node (or proto-node-list) implementation.